components/central-application-gateway/internal/proxy/cache.go

```diff
@@ -0,0 +1,76 @@
+package proxy
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	"github.com/kyma-project/kyma/components/central-application-gateway/internal/csrf"
+	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/apperrors"
+	"github.com/kyma-project/kyma/components/central-application-gateway/pkg/authorization"
+	gocache "github.com/patrickmn/go-cache"
+)
+
+const cleanupInterval = 60
+
+// CacheEntry stores information about proxy configuration in cache
+type CacheEntry struct {
+	Proxy                 *httputil.ReverseProxy
+	AuthorizationStrategy *authorizationStrategyWrapper
+	CSRFTokenStrategy     csrf.TokenStrategy
+}
+
+type authorizationStrategyWrapper struct {
+	actualStrategy authorization.Strategy
+	proxy          *httputil.ReverseProxy
+}
+
+func (ce *authorizationStrategyWrapper) AddAuthorization(r *http.Request) apperrors.AppError {
+
+	ts := func(transport *http.Transport) {
+		ce.proxy.Transport = transport
```

Comment:
Modification of shared object without a mutex

Reply:
> Modification of shared object (proxy) without a mutex

As agreed we will handle this in a separate PR.